src/frontend/src/expr/mod.rs

```diff
@@ -120,6 +120,23 @@ impl ExprImpl {
     pub fn cast_explicit(self, target: DataType) -> Result<ExprImpl> {
         FunctionCall::new_cast(self, target, CastContext::Explicit)
     }
+
+    /// Create "cast" expr to string (`varchar`) type. This is different from a real cast, as
+    /// boolean is converted to a single char rather than full word.
+    ///
+    /// Choose between `cast_output` and `cast_{assign,explicit}(Varchar)` based on `PostgreSQL`'s
+    /// behavior on bools. For example, `concat(':', true)` is `:t` but `':' || true` is `:true`.
+    /// All other types have the same behavior when formatting to output and casting to string.
+    ///
+    /// References in `PostgreSQL`:
+    /// * [cast](https://github.com/postgres/postgres/blob/a3ff08e0b08dbfeb777ccfa8f13ebaa95d064c04/src/include/catalog/pg_cast.dat#L437-L444)
+    /// * [impl](https://github.com/postgres/postgres/blob/27b77ecf9f4d5be211900eda54d8155ada50d696/src/backend/utils/adt/bool.c#L204-L209)
+    pub fn cast_output(self) -> Result<ExprImpl> {
+        if self.return_type() == DataType::Boolean {
+            return Ok(FunctionCall::new(ExprType::BoolOut, vec![self])?.into());
+        }
+        self.cast_assign(DataType::Varchar)
```

Comment:
Why do we use `cast_assign` instead of `cast_explicit` here?

Reply:
Given the target is hard-coded to `Varchar`, both `assign` and `explicit` would allow the cast. There is no difference, as the error checking inside always pass.